src/vmm/src/device_manager/mmio.rs

```diff
@@ -156,38 +156,44 @@ impl MMIODeviceManager {
         }
     }
 
-    /// Allocates resources for a new device to be added.
-    fn allocate_new_slot(&mut self, irq_count: u32) -> Result<MMIODeviceInfo> {
-        let irqs = self.irqs.get(irq_count)?;
-        let slot = MMIODeviceInfo {
-            addr: self.next_avail_mmio,
-            len: MMIO_LEN,
-            irqs,
-        };
-        self.next_avail_mmio += MMIO_LEN;
-        Ok(slot)
+    // Allocates an irq and registers it with KVM for a given VM, event.
+    fn allocate_irq(&mut self, vm: &VmFd, evt: &EventFd) -> Result<Vec<u32>> {
+        let irq = self.irqs.get(1)?;
```

Comment:
This should either
- take a `count_irqs` argument and have support to allocate and register multiple IRQs
  or
- if it's only supposed to do one irq at a time and simply be called multiple times, it should return `Result<u32>` instead of a vec of irqs